main.go

```diff
@@ -74,6 +79,15 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
```

Comment:
if you initiate the zap logger why not also pass in kube client so if anyone else uses migrations.Run function  they wont have issues

https://github.com/replicatedhq/embedded-cluster/blob/95b94dcbecd2610c8f1a593f8a30e8ae093c4da6/pkg/kubeutils/client.go#L15-L18

Reply:
All that does is discards error messages though IIRC?

Reply:
it gives me this big wall of text

```
[controller-runtime] log.SetLogger(...) was never called; logs will not be displayed.
Detected at:
	>  goroutine 1 [running]:
	>  runtime/debug.Stack()
	>  	/home/runner/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.linux-amd64/src/runtime/debug/stack.go:24 +0x5e
	>  sigs.k8s.io/controller-runtime/pkg/log.eventuallyFulfillRoot()
	>  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.3/pkg/log/log.go:60 +0xcd
	>  sigs.k8s.io/controller-runtime/pkg/log.(*delegatingLogSink).WithName(0xc000069700, {0x1ebf875, 0x14})
	>  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.3/pkg/log/deleg.go:147 +0x3e
	>  github.com/go-logr/logr.Logger.WithName({{0x25e25ed8, 0xc000069700}, 0x0}, {0x1ebf875?, 0xc0009f0470?})
	>  	/home/runner/go/pkg/mod/github.com/go-logr/logr@v1.4.1/logr.go:345 +0x36
	>  sigs.k8s.io/controller-runtime/pkg/client.newClient(0x4c5e60?, {0x0, 0x0, {0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0})
	>  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.3/pkg/client/client.go:129 +0xf1
	>  sigs.k8s.io/controller-runtime/pkg/client.New(0x1eab3cd?, {0x0, 0x0, {0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0})
	>  	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.3/pkg/client/client.go:110 +0x7d
	>  main.(*hostInfo).configureKubernetesClient(0xc000237808)
	>  	/home/runner/work/embedded-cluster/embedded-cluster/cmd/embedded-cluster/uninstall.go:115 +0xc5
	>  main.newHostInfo(_)
	>  	/home/runner/work/embedded-cluster/embedded-cluster/cmd/embedded-cluster/uninstall.go:262 +0x1ff
	>  main.init.func12(0xc000894f80)
	>  	/home/runner/work/embedded-cluster/embedded-cluster/cmd/embedded-cluster/uninstall.go:329 +0x19b
	>  github.com/urfave/cli/v2.(*Command).Run(0x26e582a0, 0xc000894f80, {0xc0002e8850, 0x1, 0x1})
	>  	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.27.2/command.go:276 +0x97d
	>  github.com/urfave/cli/v2.(*Command).Run(0xc000977340, 0xc000894ec0, {0xc0001b4000, 0x2, 0x2})
	>  	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.27.2/command.go:269 +0xbb7
	>  github.com/urfave/cli/v2.(*App).RunContext(0xc000596200, {0x25e217b8, 0xc0008303c0}, {0xc0001b4000, 0x2, 0x2})
	>  	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.27.2/app.go:333 +0x58b
	>  main.main()
	>  	/home/runner/work/embedded-cluster/embedded-cluster/cmd/embedded-cluster/main.go:41 +0x2e7
Draining node...
```

Reply:
done

Reply:
And now reverted
this hid lines like
```
2024-06-03T07:49:01Z	INFO	Reconciling installation	{"controller": "installation", "controllerGroup": "embeddedcluster.replicated.com", "controllerKind": "Installation", "Installation": {"name":"laverya-ec.c.replicated-qa.internal"}, "namespace": "", "name": "laverya-ec.c.replicated-qa.internal", "reconcileID": "391014d3-75e7-4e20-aab4-03c5d297dade"}
2024-06-03T07:49:01Z	ERROR	Reconciler error	{"controller": "installation", "controllerGroup": "embeddedcluster.replicated.com", "controllerKind": "Installation", "Installation": {"name":"laverya-ec.c.replicated-qa.internal"}, "namespace": "", "name": "laverya-ec.c.replicated-qa.internal", "reconcileID": "391014d3-75e7-4e20-aab4-03c5d297dade", "error": "failed to pre-reconcile helm charts: failed to migrate registry data: create migration job: Job.batch \"registry-data-migration\" is invalid: spec.template.spec.containers[0].image: Required value"}
```